### PR TITLE
Update to 0.11.5

### PR DESCRIPTION
**Pull request by ScottBailey**
# Changes

Changes:
- Update version from `0.10.1` to `0.11.5`.
- Remove deprecated patches.
- Update retained patches.
- Add windows/boost patch.
- Update `about` section.
- Update CMake invokations to prefer activation provided tools over shell defaults.
- Fix test for osx-arm64.
- Remove DEBUG mode from win builds.

# Review Information

## Source

https://github.com/lief-project/LIEF

## License

https://github.com/lief-project/LIEF/blob/0.11.5/LICENSE

Apache-2.0

## Upstream Changes

https://github.com/lief-project/LIEF/releases

https://lief-project.github.io/doc/stable/changelog.html#february-22-2021

11.0 potentially has breaking changes with logging.

## Issues

https://github.com/lief-project/LIEF/issues

There are a number of older issues here.

Of potential interest is a UTF-8 issue: https://github.com/lief-project/LIEF/issues/653

## Pins and Requireds

https://github.com/lief-project/LIEF/blob/551ede538abeca63a158bd7c42b6b6337c92a26e/setup.cfg#L28
py >= 36

## Testing

Author tests don't currently work in our environment. This is something that should be evaluated and fixed when we next touch this recipe.

# Closing Comments

Generally this recipe is in better shape than it originally was, though room for improvement certainly remains!

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
I would improve the recipe:
- Skip py<36 (py-lief), https://github.com/lief-project/LIEF/blob/551ede538abeca63a158bd7c42b6b6337c92a26e/setup.cfg#L28
- Add `wheel` in `host` (py-lief)
- Add pinning for setuptools >=31.0.0 in host (py-lief), https://github.com/lief-project/LIEF/blob/551ede538abeca63a158bd7c42b6b6337c92a26e/setup.py#L18
- Add `pip check` for py-lief in test/requires and test/commands
- Add `license_family: Apache`

File: recipe/meta.yaml [hunk 1]
```diff
@@ -17,49 +19,66 @@ package:
...
 
 requirements:
   build:
-    - ccache
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - pkg-config            # [not win]
-    - patch  # [not win]
-    - m2-patch  # [win]
+    - pkg-config   # [not win]
```
Comment (katietz) [inline recipe/meta.yaml]: is this required to remove pkg-config support for windows?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -17,49 +19,66 @@ package:
...
+      # string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}{{ debug }}
+      run_exports:
+        - {{ pin_subpackage('liblief', max_pin='x.x') }}
+      missing_dso_whitelist:  # [s390x]
+        - $RPATH/ld64.so.1    # [s390x] Known defect.
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - pkg-config            # [not win]
```
Comment (katietz) [inline recipe/meta.yaml]: see above

File: recipe/meta.yaml [hunk 3]
```diff
@@ -17,49 +19,66 @@ package:
...
+    script: build-py-lief.sh  # [unix]
+    script: bld-py-lief.bat   # [win]
     build:
       # string: py{{ ''.join(python.split('.')[0:2]) }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}{{ debug }}
       run_exports:
         - {{ pin_subpackage('py-lief', max_pin='x.x') }}
       ignore_run_exports:
         - python
+      missing_dso_whitelist:  # [s390x]
+        - $RPATH/ld64.so.1    # [s390x] Known defect.
```
Comment (katietz) [inline recipe/meta.yaml:81]: that is no defect, that is caused by a system dependency not covered well by conda-verify.  it is for sure no defect

**Review by katietz — CHANGES_REQUESTED** [review]

**Review by varlackc — COMMENTED** [review]
other than the previews requested changes mentioned everything else seems to be in order. Once the requested changes are made we can approve the PR.

File: recipe/meta.yaml (hunk 1 above)
Comment (ScottBailey) [reply recipe/meta.yaml]: enabled.

File: recipe/meta.yaml (hunk 3 above)
Comment (ScottBailey) [reply recipe/meta.yaml:81]: removed!

**Pull request merged**